### PR TITLE
Do MOUSEBUTTONUP event on MOUSEWHEEL

### LINKED ISSUE #2240: pygame2 does not generate MOUSEBUTTONUP events for mouse wheel
**Environment:**

- **Operating system**: Windows 10
- **Python version**: 3.7.4
- **SDL version**: 2.0.12
- **PyGame version**: 2.0.0.dev22
- **Relevant hardware**: logitech wired optical mouse (basic model)

**Current behavior:**

Rolling the mouse wheel generates MOUSEBUTTONDOWN events, but no MOUSEBUTTONUP events. It does not matter if the mouse is moving or still.

**Expected behavior:**

The same behavior as pygame 1.9.6 is expected: it generates MOUSEBUTTONUP events.

**Screenshots**

NA

**Steps to reproduce:**

1. Run the provided code in pygame 1.9.6.
2. Scroll the mouse wheel up and down.
3. See the MOUSEBUTTONUP events are printed.
4. Run the provided code in pygame 2.0.0dev22 (or previous pygame2 version).
5. Scroll the mouse wheel up and down.
6. See the MOUSEBUTTONUP events are not printed.

**Test code**

If possible add a simple test program that shows the problem described in this report.
```
import pygame
from pygame.locals import KEYDOWN, K_ESCAPE, QUIT, MOUSEBUTTONUP, MOUSEBUTTONDOWN
pygame.init()
clock = pygame.time.Clock()
screen = pygame.display.set_mode((400, 400))
while 1:
    clock.tick(30)
    for e in pygame.event.get():
        if e.type == KEYDOWN and e.key == K_ESCAPE:
                quit()
        elif e.type == QUIT:
            quit()
        elif e.type == MOUSEBUTTONDOWN:
            print('Mouse button down: {}'.format(e.button))
        elif e.type == MOUSEBUTTONUP:
            print('Mouse button up: {}'.format(e.button))
```

**Stack trace/error output/other error logs**

```
paste other relevent logs or stack traces here, if applicable
```

### PR DESCRIPTION
**Pull request by ankith26**
Probably fixes #2240

### CODE REVIEW

**Comment by ankith26** [comment]
Tested this PR locally, and it works as expected

**Comment by ankith26** [comment]
Overall, this PR makes many changes to ensure that output is much more similar to pygame 1.9.6

File: src_c/event.c [hunk 1]
```diff
@@ -157,6 +157,14 @@ _pg_remove_pending_PGS_VIDEOEXPOSE(void * userdata, SDL_Event *event)
     return 1;
 }
 
+static int
```
Comment (siggisv) [inline src_c/event.c:160]: Small note: C already has a function, `int abs(int n)`, defined in `<stdlib.h>`, that does exactly what this function is doing (see: https://en.cppreference.com/w/c/numeric/math/abs).

Comment (ankith26) [reply src_c/event.c:160]: Hmmm ok, I didn’t know it was there. Anyways, we can save an include by defining the function ourselves right ;)

Comment (illume) [reply src_c/event.c:160]: There is also SDL_abs.

**Review by illume — APPROVED** [review]
👍 thanks. Seems to work fine on my mac.

**Pull request merged**